src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/ContributionsExecutor.java

```diff
@@ -66,13 +100,87 @@ public Response count(boolean isUsersRequest) throws Exception {
           inputProcessor.getRequestUrlIfGetRequest(servletRequest));
     }
     if ("csv".equalsIgnoreCase(requestParameters.getFormat())) {
-      ExecutionUtils exeUtils = new ExecutionUtils(processingData);
+      var exeUtils = new ExecutionUtils(processingData);
       exeUtils.writeCsvResponse(results, servletResponse,
           ExecutionUtils.createCsvTopComments(URL, TEXT, Application.API_VERSION, metadata));
       return null;
     }
     return DefaultAggregationResponse.of(new Attribution(URL, TEXT), Application.API_VERSION,
         metadata, results);
   }
+  
+  /**
+   * Performs a count calculation for /users/count.
+   * 
+   * @param mapRed a MapReducer of OSM contributions
+   * @return SortedMap with counts of users aggregated by timestamp
+   * @throws Exception thrown by {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapAggregator
+   * #countUniq() countUniq}
+   * @throws UnsupportedOperationException thrown by
+   *         {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapReducer#aggregateByTimestamp()
+   *         aggregateByTimeStamp}
+   */
+  private SortedMap<OSHDBTimestamp, Integer> usersCount(MapReducer<OSMContribution> mapRed)
+      throws UnsupportedOperationException, Exception {
+    return mapRed.aggregateByTimestamp().map(OSMContribution::getContributorUserId).countUniq();
+  }
+  
+  /**
+   * Performs a count calculation for /contributions/count, /contributions/density or
+   * /contribution/latest/count.
+   * 
+   * @param mapRed a MapReducer of OSM contributions
+   * @param isContributionsLatest the boolean value relative to the endpoint /contributions/latest 
+   * @return SortedMap with counts of contributions aggregated by timestamp
+   * @throws Exception thrown by {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapAggregator
+   *         #count() count}
+   * @throws UnsupportedOperationException thrown by
+   *         {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapReducer#aggregateByTimestamp()
+   *         aggregateByTimeStamp}
+   */
+  private SortedMap<OSHDBTimestamp, Integer> contributionsCount(MapReducer<OSMContribution> mapRed,
+      boolean isContributionsLatest) throws UnsupportedOperationException, Exception {
+    if (isContributionsLatest) {
+      MapReducer<List<OSMContribution>> mapRedGroupByEntity = mapRed.groupByEntity();
+      Optional<FilterExpression> filter = processingData.getFilterExpression();
+      if (filter.isPresent()) {
+        mapRedGroupByEntity = mapRedGroupByEntity.filter(filter.get());
+      }
+      return contributionsFilter(mapRedGroupByEntity
+          .map(listContrsPerEntity -> listContrsPerEntity.get(listContrsPerEntity.size() - 1)))
+          .aggregateByTimestamp(OSMContribution::getTimestamp).count();
```

Comment:
Can you explain why the `aggregateByTimestamp` is done after the `latest` contribution is determined from the list of contributions?

I mean, the exact behaviour of the endpoint is not specified in the documentation[¹], but intuitively I would have expected that if one requests multiple time intervals, that each one is handled individually. i.e. that the output of `/contributions/count/latest?…&time=t1,t2,t3` is the "concatenation" of the out two requests with `time=t1,t2` and `time=t2,t3`. The current implementation would be instead "distribute" the total result (i.e. `time=t1,t3`) into the two intervals from `time=t1,t2,t3`.

[¹] maybe we should do that?

Reply:
I considered the endpoint with respect to entities (latest among all contributions of an entity in the given interval). Otherwise, given one entity having some contributions in two time intervals, the output will have the latest contribution in the first interval for the entity and another latest contribution for the same entity in the second interval.